### DISCUSSION TITLE
passing list data structure in channel

### DISCUSSION THREAD

**Discussion by ntp3105** [adamreeve/npTDMS#330] [Q&A] [answered]
Can I pass multiple channel names to channel ? For instance,
channel = group['channel_1', 'channel_2']

**Comment by adamreeve** [comment]
Hi @ntp3105, no you can't, and I don't understand what result you expect to get back. What are you trying to achieve?

**Reply by ntp3105** [reply]
Thank you for the reply, @adamreeve 
For my query, I realize I can add for loop  over channel names instead of calling all of them at once. As for what I am trying to achieve, I have lots of tdms files with different group and channel names. I don't know the names of the channels and group mentioned in the files. Can I use npTDMS to read them from the file?

**Reply by adamreeve** [reply, marked as answer]
Right, yes using a loop over names would make sense. If you don't know the group and channel names ahead of time you can loop over all groups and channels using the `TdmsFile.groups` and `TdmsGroup.channels` methods. Eg:
```python
for group in tdms_file.groups():
    print(f"group '{group.name}'")
    for channel in group.channels():
        print(f"channel '{channel.name}'")
```

Or alternatively you can actually iterate over the `TdmsFile` and `TdmsGroup` objects to get the names too:
```python
for group_name in tdms_file:
    print(f"group '{group_name}'")
    group = tdms_file[group_name]
    for channel_name in group:
        print(f"channel '{channel_name}'")
        channel = group[channel_name]
```

**Reply by ntp3105** [reply]
Thank you so much @adamreeve. your reply helped me build code for extracting data from tdms files.
